dwave/embedding/polynomialembeddor.py

```diff
@@ -0,0 +1,1222 @@
+        """Computes the score of a partial native clique embedding given the score
+        attained by the already-placed ells, together with the ell block
+        defined by ``hbar = (y0,xmin,xmax)``, and  ``vbar = (x0,ymin,ymax)``.
+
+        In the plain :class:`eden_processor` class, this is simply the number of ells
+        contained in the partial native clique after adding the new ells.
+        """
+        (y0, xmin, xmax) = hbar
+        (x0, ymin, ymax) = vbar
+        if rscore is None:
+            rscore = 0
+        hscore = self.hline_score(y0, xmin, xmax)
+        vscore = self.vline_score(x0, ymin, ymax)
+        if vscore < hscore:
+            score = rscore + vscore
+        else:
+            score = rscore + hscore
+        return score
+
+    def maxCliqueWithRectangle(self, R, maxCWR):
+        """This does the dirty work for :func:`nativeCliqueEmbed`.  Not meant to be
+        understood or called on its own.  Guaranteed to maintain the inductive
+        hypothesis that ``maxCWR`` is optimal.  We put in the tiniest amount of
+        effort to return a uniform random choice of a maximum-sized native
+        clique embedding.
+
+        INPUTS:
+            R (tuple): the rectangle specified as a tuple ``(xmin,xmax,ymin,ymax)``
+
+            maxCWR (dict): the dictionary we're building inductively which maps
+            ``R -> (score, ell, r, num)`` where score is the size of the best
+            clique with working rectangle ``R``, and ``ell`` and ``r`` indicate how to
+            reconstruct that clique: add the maximum line bundle in ``ell``, and
+            recursively examine the working rectangle ``r``.  The parameter ``num`` is the number
+            of cliques with ``R`` as a working rectangle and size equal to ``score``.
+
+        OUTPUT:
+            score (int): the score for the returned clique (just ``len(clique)``
+            in the class :class:`eden_processor`; may differ in subclasses)
+
+            best (tuple): a tuple ``(score, ell, parent, num)`` to be stored in
+            ``maxCWR[R]``.
+
+            * score: as above
+            * ell: ``(x0,x1,y0,y1)`` defines the unit cells in an ell-shaped
+                region: ``(x0,y1),...,(x0,y0),...,(x1,y0)``
+            * parent: the rectangle ``R1`` for which the clique generated
+                recursively by looking up ``maxCWR[R1]`` as described above.
+            * num: the number of partial native clique embeddings with ``R``
+                as a working rectangle and a score of ``score``.
+
+        """
+
+        (xmin, xmax, ymin, ymax) = R
+        best = nothing = 0, None, None, 1
+        bestscore = None
+        count = 0
+        N = self.N
+
+        Xlist = (xmin, xmax, xmin + 1, xmax), (xmax, xmin, xmin, xmax - 1)
+        Ylist = (ymin - 1, ymax, ymin - 1,
+                 ymax), (ymax + 1, ymin, ymin, ymax + 1)
+        XY = [(X, Y) for X in Xlist for Y in Ylist if 0 <= Y[2] <= Y[3] < N]
```

Comment:
Lowercase or snake case? If I saw `XY` without context, I would have expected it to be a constant.